File: Programmers/N개의 최소공배수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;


int solution(vector<int> arr) {
	int answer = arr[arr.size() - 1];
	sort(arr.begin(), arr.end());
    bool flag = 1;
	while (flag)
	{
		for (int i : arr) {
			if (answer % i != 0)
			{
				answer++;
				break;
			}
            else if(answer % i == 0 && i == arr[arr.size()-1])
                flag = 0;
		}
	}
	return answer;
}

File: Programmers/N개의 최소공배수.java
import java.util.*;

class Solution {
    public int solution(int[] arr) {
        Arrays.sort(arr);
        
        int answer = arr[arr.length - 1];
        boolean flag = true;
        
        while(flag)
        {
            for(int i : arr)
                if(answer % i != 0) {
                    answer++;
                    break;
                }
                else if(answer % i == 0 && i == arr[arr.length - 1])
                {
                    flag = false;
                    break;
                }
        }
        return answer;

    }
}